src/JMS/Serializer/Metadata/Driver/XmlDriver.php

```diff
@@ -82,6 +82,8 @@ protected function loadMetadataFromFile(\ReflectionClass $class, $path)
             $discriminatorMap[(string) $entry->attributes()->value] = (string) $entry;
         }
 
+        $strict_deserialize = 'false' == (string)$elem->attributes()->{'discriminator-strict-deserialize'} ? false : true;
```

Comment:
here is enough `'false' !== (string)$elem->attributes()->{'discriminator-strict-deserialize'}`, no `? false : true` is needed